zeppelin-server/src/main/java/org/apache/zeppelin/server/ZeppelinServer.java

```diff
@@ -364,6 +364,7 @@ public void shutdown(int exitCode) {
         if (sharedServiceLocator != null) {
           if (!zConf.isRecoveryEnabled()) {
             sharedServiceLocator.getService(InterpreterSettingManager.class).close();
+            sharedServiceLocator.shutdown();
```

Comment:
Yes, this is the simple variant, but it actually belongs outside the ‘if’ statement.